src/components/signup.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { Fragment } from 'react';
+import '../styles/signup.scss';
+import '../styles/button.scss';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+
+class Signup extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: '',
+      lastName: '',
+      username: '',
+      email: '',
+      password: '',
+      errorMessage: '',
+    };
+  }
+
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  handleSubmit = (event) => {
+    const { firstName, lastName, username, email, password } = this.state;
+    const { history } = this.props;
+    event.preventDefault();
+    axios
+      .post(`${process.env.API_URL}/users/signup`, {
+        first_name: firstName,
+        last_name: lastName,
+        username,
+        email,
+        password,
+      })
+      .then(() => {
+        history.push('/login');
+      })
+      .catch((error) => {
+        this.setState({ errorMessage: error.response.data.message });
```

Comment:
this error message will never be displayed because you immediately redirect to `/`